core/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -356,6 +356,15 @@ trait OptimisticTransactionImpl extends TransactionalWrite
     updateMetadataInternal(_metadata, ignoreDefaultProperties)
   }
 
+  /**
+   * This updates the protocol for the table to a set newProtocol
+   */
+  def updateProtocol(newProtocolOption: Option[Protocol]): Unit = {
+    if (newProtocolOption != None) {
```

Comment:
I am surprised that there were no existing protocol updating method in opttxn, and this DDL is the first one requiring it. 

@xupefei could you double check this protocol upgrade path, since I dont if there is any complexity with regards to table features.